sklearn/neighbors/lof.py

```diff
@@ -0,0 +1,313 @@
+
+        - from scipy.spatial.distance: ['braycurtis', 'canberra', 'chebyshev',
+          'correlation', 'dice', 'hamming', 'jaccard', 'kulsinski',
+          'mahalanobis', 'matching', 'minkowski', 'rogerstanimoto',
+          'russellrao', 'seuclidean', 'sokalmichener', 'sokalsneath',
+          'sqeuclidean', 'yule']
+
+        See the documentation for scipy.spatial.distance for details on these
+        metrics.
+
+    metric_params : dict, optional (default = None)
+        Additional keyword arguments for the metric function.
+
+    contamination : float in (0., 0.5), optional (default=0.1)
+        The amount of contamination of the data set, i.e. the proportion
+        of outliers in the data set. When fitting it is used to define the
+        threshold on the decision function.
+
+    n_jobs : int, optional (default = 1)
+        The number of parallel jobs to run for neighbors search.
+        If ``-1``, then the number of jobs is set to the number of CPU cores.
+        Affects only :meth:`k_neighbors` and :meth:`kneighbors_graph` methods.
+    """
+    def __init__(self, n_neighbors=5, algorithm='auto', leaf_size=30,
+                 metric='minkowski', p=2, metric_params=None,
+                 contamination=0.1, n_jobs=1):
+        self._init_params(n_neighbors=n_neighbors,
+                          algorithm=algorithm,
+                          leaf_size=leaf_size, metric=metric, p=p,
+                          metric_params=metric_params, n_jobs=n_jobs)
+
+        self.contamination = contamination
+
+    def fit(self, X, y=None):
+        """Fit the model using X as training data
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix, BallTree, KDTree}
+            Training data. If array or matrix, shape [n_samples, n_features],
+            or [n_samples, n_samples] if metric='precomputed'.
+        """
+        if not (0. < self.contamination <= .5):
+            raise ValueError("contamination must be in (0, 0.5]")
+
+        super(LocalOutlierFactor, self).fit(X)
+
+        self._k_distance_value_fit_X_, self.neighbors_indices_fit_X_ = (
+            self._k_distance(None))
+
+        # Compute decision_function over training samples to define threshold_:
+        self.decision_function_train_ = -self._local_outlier_factor()
+        # as bigger is better (less abnormal)
+
+        self.threshold_ = -scoreatpercentile(
+            -self.decision_function_train_, 100. * (1. - self.contamination))
+
+        # XXX May be optimized (if X is not None) by only computing it for
+        # X_sub_samples = neighbors of neighbors of X ?
+        return self
+
+    def predict(self, X=None):
+        """Predict Local Outlier Factor of X.
+
+        The local outlier factor (LOF) of a sample captures its
+        supposed `degree of abnormality'.
+        It is the average of the ratio of the local reachability density of
+        a sample and those of its k-nearest neighbors.
+
```

Comment:
The predict method predicts the labels of the samples (1 inlier, -1 outlier), which is what is returned, rather than the Local Outlier Factor which is a score and corresponds to the decision_function method. Reading the description of the predict method, one could think that it predicts the scores.